nncf/quantization/algorithms/channel_alignment/algorithm.py

```diff
@@ -0,0 +1,379 @@
+from nncf.common.utils.backend import BackendType
+from nncf.common.utils.backend import get_backend
+from nncf.quantization.algorithms.algorithm import Algorithm
+from nncf.quantization.algorithms.channel_alignment.backend import ConvParamsContainer
+from nncf.quantization.algorithms.channel_alignment.backend import DimsDescriptor
+from nncf.quantization.algorithms.fast_bias_correction.backend import ALGO_BACKENDS
+
+TModel = TypeVar("TModel")
+TTensor = TypeVar("TTensor")
+
+FAST_BIAS_CORRECTION_THRESHOLD = 2
+
+
+class ChannelAlignment(Algorithm):
+    """
+    Post-training FastBiasCorrection algorithm implementation.
+
+    The main purpose of this algorithm to reduce quantization error
+    via correction the bias of the Convolutions, FullyConnected, etc. layers.
+    The algorithm pipeline is very simple:
+        - we collects floating-point statistics from the corresponding model for the layers with bias;
+        - then we gets the quantized model and try to reduce it's error by correction of the bias;
+        - the shift calculates using the sub-graph that consists of the correction layer and
+        weight quantizer-dequantizer pair or fake quantize node;
+        - the floating-point statistics uses as input for
+        the sub-graph and further quantization output calculation;
+        - in the end we corrects the original bias by the difference (shift)
+        between floating-point and quantized outputs.
+    """
+
+    def __init__(
+        self,
+        subset_size: int = 100,
+        inplace_statistics: bool = True,
+        backend_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param subset_size: Size of a subset for the statistics collection,
+            defaults to 100.
+        :param threshold: The magnitude threshold that regulates the application of the
+            shift. Magnitude calculates as the maximum of the absolute ratio of the
+            shift to the original bias value. If the calculated value is less than the
+            threshold, the shift will apply to the bias, defaults to 2.
+        :param apply_for_all_nodes: If True, then the bias correction be applied to all
+            quantized nodes, if the node has no bias then a bias node will be inserted,
+            and if False, then the bias correction will only be applied to quantized
+            nodes that have a bias.
+        :param inplace_statistics: Defines wheather to calculate quantizers statistics
+            by backend graph operations or by default Python implementation, defaults
+            to True.
+        :param backend_params: Backend specific parameters.
+        """
+        super().__init__()
+        self.subset_size = subset_size
+        self.inplace_statistics = inplace_statistics
+        self.backend_params = backend_params
+        self.nncf_graph = None
+        self._backend_entity = None
+        self._nncf_grpah = None
+        self._q = 1e-4
```

Comment:
what is `_q`?

Reply:
q stands for quantile

Reply:
It's about add docstring or use not onechar name

Reply:
I made clearer name for this param, please check